### PR TITLE
Remove os_utils module and fix deprecations

### PR DESCRIPTION
**Pull request by jgersak**

### CODE REVIEW

File: sentinelhub/io_utils.py [hunk 1]
```diff
@@ -166,6 +165,13 @@ def read_numpy(filename: str) -> np.ndarray:
     return np.load(filename)
 
 
+def _create_parent_folder(filename: str) -> None:
+    """Create parent folder for input filename recursively"""
```
Comment (zigaLuksic) [inline sentinelhub/io_utils.py:169]: what is recursive about this?

Comment (jgersak) [reply sentinelhub/io_utils.py:169]: `os.makedirs` is recursive directory creation function that makes all intermediate-level directories needed to contain the leaf directory 
docstring could be removed add almost zero meaning

Comment (zigaLuksic) [reply sentinelhub/io_utils.py:169]: yes, in this case it's more confusing than explanatory.

Comment (jgersak) [reply sentinelhub/io_utils.py]: removed

**Pull request merged**